### ISSUE TITLE
Mismatched found in ensure_batch_predictions_are_consistent. Is is important?

### ISSUE THREAD

**Issue by huhk-sysu** [allenai/allennlp#5692]
Hi, I'm using `allennlp.common.testing.model_test_case` to test my model, and it reports lots of mismatched, though the value seems small.
![image](https://user-images.githubusercontent.com/14769033/179471352-982af8a9-4a62-4bdd-8214-b0cbfe23fb4f.png)

1. I'm not sure if it is important to me. Do I need to fix it?
2. If so, what would be the cause and how can I fix it?

 ps: I notice that my prediction (which are calculated based on the logits) are the same.
